pwndbg/color/memory.py

```diff
@@ -26,12 +27,27 @@
     ],
 )
 
+pwndbg.config.add_param(
+    "telescope-free-heap-bins",
+    False,
+    "whether or not to resolve heap addresses to bin names while telescoping",
+)
+
 
-def get_address_and_symbol(address: int) -> str:
+def get_address_and_symbol(address: int, display_heap_free_bins=True) -> str:
     """
     Convert and colorize address 0x7ffff7fcecd0 to string `0x7ffff7fcecd0 (_dl_fini)`
     If no symbol exists for the address, return colorized address
     """
+
+    # Attempt to resolve it as a free chunk
+    if display_heap_free_bins and pwndbg.config.telescope_free_heap_bins:
+        if (page := pwndbg.gdblib.vmmap.find(address)) is not None and "[heap" in page.objfile:
```

Comment:
I understand this is for efficiency, i.e. don't call `heap_freebin_address_lookup()` if `address` isn't in a mapping labelled "[heap]".

As I recall, pwndbg only marks the GLIBC sbrk heap with "[heap]", so `heap_freebin_address_lookup()` may not run when there are other heaps present.

Reply:
> this would not work with multiple heaps that can be there when multiple threads are in play.

Exactly.

GLIBC malloc may also create multiple heaps in a single-threaded environment if sbrk fails due to memory pressure.